service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +189,212 @@ void updatePipelineWorkspaceId() {
+
+    // add a pipeline input to the imputation pipeline
+    PipelineInputDefinition newInput =
+        new PipelineInputDefinition(pipeline.getId(), "new_integer_input", "Integer", true);
+
+    pipelineInputDefinitionsRepository.save(newInput);
+
+    Object missingRequiredInputs =
+        new LinkedHashMap<String, Object>(Map.of("not_an_input", "who cares"));
+    ValidationException exception =
+        assertThrows(
+            ValidationException.class,
+            () -> pipelinesService.validateInputs(pipelinesEnum, missingRequiredInputs));
+    assertEquals(
+        "Problems with pipelineInputs: multi_sample_vcf is required; new_integer_input is required",
+        exception.getMessage());
+  }
+
+  @Test
+  void validateInputsOneMissingOneWrongType() {
+    PipelinesEnum pipelinesEnum = PipelinesEnum.IMPUTATION_BEAGLE;
+    Pipeline pipeline = pipelinesRepository.findByName(pipelinesEnum.getValue());
+
+    // add a pipeline input to the imputation pipeline
+    PipelineInputDefinition newInput =
+        new PipelineInputDefinition(pipeline.getId(), "new_integer_input", "Integer", true);
+
+    pipelineInputDefinitionsRepository.save(newInput);
+
+    Object missingRequiredInputs =
+        new LinkedHashMap<String, Object>(Map.of("new_integer_input", "this is not an integer"));
+    ValidationException exception =
+        assertThrows(
+            ValidationException.class,
+            () -> pipelinesService.validateInputs(pipelinesEnum, missingRequiredInputs));
+    assertEquals(
+        "Problems with pipelineInputs: multi_sample_vcf is required; new_integer_input must be of type Integer",
+        exception.getMessage());
+  }
+
+  @Test
+  void validateInputsNotAMap() {
+    PipelinesEnum pipelinesEnum = PipelinesEnum.IMPUTATION_BEAGLE;
+    Object inputs = new ArrayList<>(List.of("this", "is", "not", "a", "map"));
+    assertThrows(
+        ValidationException.class, () -> pipelinesService.validateInputs(pipelinesEnum, inputs));
+  }
+
+  @Test
+  void validateRequiredInputPresent() {
+    PipelineInputDefinition requiredInput =
+        new PipelineInputDefinition(1L, "required_input", "Integer", true);
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(requiredInput));
+
+    LinkedHashMap<String, Object> inputs =
+        new LinkedHashMap<>(Map.of("required_input", "this is a string"));
+
+    // even though it's the wrong type, the required input is present, so should not return an error
+    // message
+    assertTrue(pipelinesService.validateRequiredInputs(inputDefinitions, inputs).isEmpty());
+  }
+
+  private static Stream<Arguments> inputTypeValidationsValid() {
+    return Stream.of(
+        // arguments are: type specification, value to be tested, isRequired
+        arguments("Integer", 123, true),
+        arguments("Integer", "123", true),
+        arguments("String", "I am a string", true),
+        arguments("VCF", "path/to/file.vcf.gz", true),
```

Comment:
yeah, agreed. I can easily add stripping now but I'll wait on the scheme validation bit until we know more about where the inputs are going to be coming from

Reply:
actually i'm going to wait on stripping as well, since it's not needed for validation but will be needed once we're using the values (i.e. when we create the cbas submission object). I'll add a note in that ticket to implement stripping